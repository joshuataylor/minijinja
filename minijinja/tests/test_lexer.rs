#![cfg(feature = "unstable_machinery")]
use minijinja::machinery::tokenize;

use std::fmt::Write;

#[test]
fn test_lexer() {
    insta::glob!("lexer-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let tokens: Result<Vec<_>, _> = tokenize(&contents, false).collect();
        let tokens = tokens.unwrap().into_iter().map(|x| x.0).collect::<Vec<_>>();
        insta::with_settings!({
            description => contents.trim_end(),
            omit_expression => true
        }, {
            let mut stringified = String::new();
            for token in tokens {
                writeln!(stringified, "{:?}", token).unwrap();
            }
            insta::assert_snapshot!(&stringified);
        });
    });
}
