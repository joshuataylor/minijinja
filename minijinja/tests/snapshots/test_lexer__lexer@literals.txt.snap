---
source: minijinja/tests/test_lexer.rs
description: "{{ true }}\n{{ false }}\n{{ null }}\n{{ 1.0 }}\n{{ 2 }}\n{{ [1, 2, 3] }}\n{{ {\"foo\": \"bar\"} }}"
input_file: minijinja/tests/lexer-inputs/literals.txt
---
[
    VARIABLE_START(false),
    IDENT(true),
    VARIABLE_END(false),
    TEMPLATE_DATA("\n"),
    VARIABLE_START(false),
    IDENT(false),
    VARIABLE_END(false),
    TEMPLATE_DATA("\n"),
    VARIABLE_START(false),
    IDENT(null),
    VARIABLE_END(false),
    TEMPLATE_DATA("\n"),
    VARIABLE_START(false),
    FLOAT(1.0),
    VARIABLE_END(false),
    TEMPLATE_DATA("\n"),
    VARIABLE_START(false),
    INT(2),
    VARIABLE_END(false),
    TEMPLATE_DATA("\n"),
    VARIABLE_START(false),
    BRACKET_OPEN,
    INT(1),
    COMMA,
    INT(2),
    COMMA,
    INT(3),
    BRACKET_CLOSE,
    VARIABLE_END(false),
    TEMPLATE_DATA("\n"),
    VARIABLE_START(false),
    BRACE_OPEN,
    STR("foo"),
    COLON,
    STR("bar"),
    BRACE_CLOSE,
    VARIABLE_END(false),
    TEMPLATE_DATA("\n"),
]
